### PR TITLE
doc: Add section about testing

### PR DESCRIPTION
**Pull request by yaqwsx**

### CODE REVIEW

**Comment by Trphajzlik** [comment]
Za mě pěkné a srozumitelné.

File: doc/intro/index.rst [hunk 1]
```diff
@@ -16,4 +16,5 @@ Then proceed to the following sections:
 
 .. toctree::
     compiling
-    structure
\ No newline at end of file
+    structure
+    testing
```
Comment (Ondraceq) [inline doc/intro/index.rst:20]: Add newline (less diffs in the future)

File: doc/intro/testing.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
+Writing and running tests
+=========================
+
+The project features a number of unit tests. If you develop a new part of the
+project, your are encouraged to write them. Not only it makes your life easier
+(as it is much easier to code and debug against test), but you also ensure that
```
Comment (Ondraceq) [inline doc/intro/testing.rst:6]: against `tests` (or `a test`)

File: doc/intro/testing.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    rmake --test <pathName>...       Compile and run tests from given path prefix
+    rmake --test <suitename>...      Compile and run tests from given suite
+
+
+The command will compile and run specified tests. Also, the tab-completion shows
+you the available options. Note that only tests from configured suites are
+available.
+
+You can specify the test based on the same criterion as `compilation targets
+<compiling.html#compilation-targets>`__. So you can run test for the whole
```
Comment (Ondraceq) [inline doc/intro/testing.rst:31]: run `tests`

File: doc/intro/testing.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+suite, given test or all tests in the specified path prefix. For the sake of
```
Comment (Ondraceq) [inline doc/intro/testing.rst:32]: given `a` test

File: doc/intro/testing.rst [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+simplicity, you currently cannot pass any arguments to the tests. If you need
+so, you can always invoke the test executable directly.
```
Comment (Ondraceq) [inline doc/intro/testing.rst:34]: if you need `to`/if you need `to do so`

File: doc/intro/testing.rst [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+And that's everything you need. Now you can just write test source files in the
+`test` directory. See `Catch2 tutorial
+<https://github.com/catchorg/Catch2/blob/v2.x/docs/tutorial.md>`__ for more
+details.
+
+When your software module is not header-only, it makes sense to extract it into
+a static library and link that library to your executable and tests. It will
+significantly speed-up the compilation as you don't have to compile all sources
+twice.
```
Comment (Ondraceq) [inline doc/intro/testing.rst:80]: no newline

File: doc/intro/testing.rst [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+-------------
+
+For the purposes of RoFI, a test is an executable in ``PATH`` that satisfies the
+following:
+
+-  its name starts with the prefix `test-`,
+-  it has an associated CMake target in one of the projects,
+-  it can run with no command line arguments,
+-  if and only if the tests fails, the return code is non-zero, and
+-  it gives user-readable output on the stdout indicating what went wrong in the
```
Comment (Ondraceq) [inline doc/intro/testing.rst:46]: Doesn't Catch use stderr as well?

File: doc/intro/testing.rst [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    file(GLOB TEST_SRC test/*.cpp)
+    add_executable(test-coolSoftwareModule ${TEST_SRC})
+    target_link_libraries(test-coolSoftwareModule PRIVATE Catch2::Catch2)
+
+And that's everything you need. Now you can just write test source files in the
+`test` directory. See `Catch2 tutorial
+<https://github.com/catchorg/Catch2/blob/v2.x/docs/tutorial.md>`__ for more
+details.
+
+When your software module is not header-only, it makes sense to extract it into
```
Comment (Ondraceq) [inline doc/intro/testing.rst:77]: `if` your software...

**Comment by yaqwsx** [comment]
Díky, připomínky zapracovány.

**Review by Ondraceq — APPROVED** [review]
Líbí se mi to

**Pull request merged**